controller/src/main/java/org/jboss/as/controller/services/path/PathInfoHandler.java

```diff
@@ -0,0 +1,428 @@
+import java.util.Calendar;
+import java.util.List;
+import java.util.Locale;
+import java.util.ResourceBundle;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+import org.jboss.as.controller.AbstractRuntimeOnlyHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.ObjectTypeAttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationDefinition;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.SimpleOperationDefinitionBuilder;
+import org.jboss.as.controller.client.helpers.MeasurementUnit;
+import org.jboss.as.controller.descriptions.StandardResourceDescriptionResolver;
+import org.jboss.as.controller.descriptions.common.ControllerResolver;
+import org.jboss.as.controller.logging.ControllerLogger;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.controller.registry.AttributeAccess;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.wildfly.security.manager.WildFlySecurityManager;
+
+/**
+ * Handler for file usage metric which contains the total size of a folder and the usable space (as in Java nio).
+ *
+ * @author <a href="mailto:ehugonne@redhat.com">Emmanuel Hugonnet</a> (c) 2015 Red Hat, inc.
+ *
+ * @see java.nio.file.FileStore#getUsableSpace()
+ */
+public class PathInfoHandler extends AbstractRuntimeOnlyHandler {
+
+    private static final DateTimeFormatter DATE_FORMAT = new DateTimeFormatterBuilder().appendInstant().appendZoneId().toFormatter(Locale.ENGLISH);
+    private static final ZoneId ZONE_ID = ZoneId.of(Calendar.getInstance().getTimeZone().getID());
+    private static final String USED_SPACE = "used-space";
+    private static final String AVAILABLE_SPACE = "available-space";
+    private static final String RESOLVED_PATH = "resolved-path";
+    private static final String LAST_MODIFIED = "last-modified";
+    private static final String CREATION_TIME = "creation-time";
+    private static final String OPERATION_NAME = "path-info";
+    private static final String FILESYSTEM = "filesystem";
+
+    private static final AttributeDefinition UNIT_ATTRIBUTE = SimpleAttributeDefinitionBuilder
+                        .create(UNIT, ModelType.STRING, true)
+                        .setAllowedValues(BYTES.getName(), KILOBYTES.getName(), MEGABYTES.getName(), GIGABYTES.getName(),
+                                TERABYTES.getName(), PETABYTES.getName(), BITS.getName(), KILOBITS.getName(),
+                                MEGABITS.getName(), GIGABITS.getName(), TERABITS.getName(), PETABITS.getName())
```

Comment:
You can use EnumValidator for this, which via a hack also results in allowed-values being set. That validator is also forgiving about case sensitivity. If you do that though when sizeUnit = MeasurementUnit.valueOf(unitModelNode.asString()) you have to convert the string to upper case.